### ISSUE TITLE
all predicted values are “no_relation”, and the Precision  value is 100%, the Recall value is 0, and the value of dev_f1 is also 0.

### ISSUE THREAD

**Issue by CV2014** [qipeng/gcn-over-pruned-trees#9]
When I am resurrecting, all predicted values are “no_relation”, and the Precision  value is 100%, the Recall value is 0, and the value of dev_f1 is also 0.
Evaluating on dev set...
['no_relation', 'no_relation', 'no_relation', 'no_relation', 'no_relation', 'no_relation', 'no_relation', 'no_relation', 'no_relation', 'no_relation', 'no_relation', 'no_relation', 'no_relation', 'no_relation', 'no_relation', 'no_relation', 'no_relation', 'no_relation', 'no_relation', 'no_relation']
Precision (micro): 100.000%
   Recall (micro): 0.000%
       F1 (micro): 0.000%
epoch 19: train_loss = 0.852363, dev_loss = 5.416558, dev_f1 = 0.0000
model saved to ./saved_models/00/checkpoint_epoch_19.pt

**Comment by dterg** [comment]
This is simply because the training data is too small. Grouping the relations into one positive relations class to increase the per-class samples, reducing the learning rate to 0.003 and choosing adam as optimizer got me ~50% F-score. Polluting the training set with some dev samples will get you a high dev F-score. Obviously not what one should do but confirms its nothing buggy - just size of the sample data.

**Comment by CV2014** [comment]
However, when I use a large corpus, the learning rate is also reduced to 0.003, and the predicted result is still the same.

**Comment by liuzhencheng** [comment]
i want know too

**Comment by qipeng** [comment]
If you're just using the sample data to train the model (included in the repo), that is probably not going to be enough. Like @dterg mentioned, training the model on a larger dataset would probably help. (Also the dev loss you are observing is probably another clue that the model is overfitting to the training set)

The 100% precision is simply a quirk in our evaluation code [here](https://github.com/qipeng/gcn-over-pruned-trees/blob/master/utils/scorer.py#L54), and should be disregarded when F1 is zero.
